src/scraping/scrape_activities.py

```diff
@@ -0,0 +1,158 @@
+"""
+Get text data from the BBC's Tiny Happy People website about kids activities
+
+Usage:
+
+First navigate to the scraping folder
+$ cd src/scraping
+
+Then run the script
+$ poetry run python scrape_activities.py
+
+The final output file is stored in data/tiny_happy_people - final.csv
+with the following columns:
+    - CONTENT: Title of the activity
+    - SHORT DESCRIPTION: Short description of the activity
+    - Age Range (if applicable): Age range of the activity
+    - Type: Type of the activity (there are 20+ different types used by BBC)
+    - URL: URL of the activity
+    - text: Scraped text, describing the activity in greater detail
+
+"""
+
+import csv
+import logging
+
+from pathlib import Path
+from time import sleep
+
+import bs4
+import pandas as pd
+import requests
+
+from bs4 import BeautifulSoup
+from tqdm import tqdm
+
+
+# URLs to scrape
+URL_PATH = Path("data/tiny_happy_people_urls - links.csv")
+# Interim file to store the scraped text
+SCRAPED_PATH = URL_PATH.parent / "tiny_happy_people - scraped.csv"
+# File to store the final table
+OUTPUTS_PATH = URL_PATH.parent / "tiny_happy_people - final.csv"
+
+# Possible headers signalling the end of the article
+END_MARKERS = [
+    "In case you missed it",
+    "Find another activity",
+]
+
+# Boilerplate text to be removed
+SCRAP_TEXT = [
+    "TwitterFacebookMessengerWhatsAppShareShare this withTwitterFacebookMessengerWhatsAppCopy linkRead more about sharing",
+    "previousnext",
+]
+
+
+def pad_element(element: bs4.element.Tag) -> str:
+    """Pad an element's content with spaces for certain tags (a recursive function)
+
+    Args:
+        element (bs4.element.Tag): HTML element
+
+    Returns:
+        str: Text with spaces
+    """
+    if isinstance(element, str):
+        return element
+    elif element.name in ["p", "h1", "h2", "h3", "h4", "h5", "h6"]:
+        return " " + " ".join(pad_element(child) for child in element.children) + " "
+    else:
+        return "".join(pad_element(child) for child in element.children)
+
+
+# Function to scrape the web page
+def web_scraper(url: str) -> str:
+    """Scrape a web page and return the content
+
+    Args:
+        url
+            URL of the web page
+
+    Returns:
+        Dataframe with URL and content
+    """
+    # Fetch webpage
+    response = requests.get(
+        url,
+        timeout=10,
+        headers={
+            "User-Agent": "Data collection for the purpose of research. For questions, reach out to karlis.kanders@nesta.org.uk"
+        },
```

Comment:
Slight nitpicking but we could remove the hardcoded parameters by adding the `timeout` as an argument to the function and moving the header as a global variable, along with `END_MARKERS`, etc